pkg/frontend/util/fake/kubernetes_client.go

```diff
@@ -0,0 +1,27 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"errors"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+)
+
+// NewFakeKubernetesClient is a wrapper for fake.NewSimpleClientset
+// which returns (kubernetes.Interface, error) instead of *k8sfake.Clientset,
+// which allows it to be used as a kubernetesClientFactory
+func NewFakeKubernetesClient(args ...interface{}) (kubernetes.Interface, error) {
```

Comment:
This will be required to add a unit test for Must Gather because [it needs a Kubernetes client](https://github.com/Azure/ARO-RP/blob/c76b5d1549bbc83c41073f926e5fcbfa398ffe10/pkg/frontend/kubeactions/mustgather.go#L30).

This and `NewKubernetesClient` in `frontend.go` stem from the fact that there can be no `type kubernetesClientFactory func` which could be satisfied by both the real and fake Clientset functions.